pkg/controller/job/job_controller.go

```diff
@@ -170,7 +170,7 @@ func NewController(podInformer coreinformers.PodInformer, jobInformer batchinfor
 
 	jm.updateStatusHandler = jm.updateJobStatus
 	jm.patchJobHandler = jm.patchJob
-	jm.syncHandler = jm.syncJob
+	jm.syncHandler = controller.SyncAndRecordWithBool("job", jm.syncJob)
```

Comment:
We already have a metric specific for the job controller https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/metrics/metrics.go#L43

Is it worth having 2 metrics? I think we should standardize the name, but leave it up to the controller to add the labels they might need.

Reply:
I am not sure what you are suggesting, would you elaborate?

Reply:
I am suggesting that this change is reverted for the job controller, as it already defines a metric with the same intent.

But we can standardize the name so that it's a consistent experience for people that need these metrics.

Reply:
got it, thanks!

Reply:
cc @soltysh for thoughts.

Reply:
@logicalhan wdyt?

I don't think it's good if we have 2 metrics that provide very similar output

Reply:
I like having the generic controller metrics, I think it is generally a step in the right direction. Are you saying there are already similar metrics for the scheduler?

Reply:
no, for the job controller. There is already a metric that has the same intent, but has domain-specific labels
https://github.com/kubernetes/kubernetes/blob/6f896dec4f45ffb82491bc9ce2393e7452886562/pkg/controller/job/metrics/metrics.go#L30-L44

Reply:
how much structure can we put in "additional_data"?

Reply:
I was thinking it could just be a catch-all additional label, so for jobs it would take the place of `completion_mode`.

Reply:
I mean what's the point of a generic controller metric if you are going to special case away from it every time...

Reply:
i'm +1 on adding `"action"` and `"additional_data"`. I would see usage of `"additional_data"` in the job controller having values like `"completion_mode=NonIndexed"`. That said, I am not deeply nuanced in my understanding of what values metric labels are expected to have; is what I described valid?

Reply:
i guess an alternative to `additional_data` might be to have specific metrics for additional information specific to individual controllers. Again, I'm not deeply nuanced on how things like single table inheritance apply to metrics, just 'thinking aloud'.

Reply:
What is the final decision here? Are we proceeding with duplicate metrics?